src/include/platform/FailSafeContext.h

```diff
@@ -55,12 +55,20 @@ class FailSafeContext
         return (accessingFabricIndex == mFabricIndex);
     }
 
-    inline bool NocCommandHasBeenInvoked() { return mNocCommandHasBeenInvoked; }
+    inline bool NocCommandHasBeenInvoked() { return mAddNocCommandHasBeenInvoked || mUpdateNocCommandHasBeenInvoked; }
+    inline bool AddNocCommandHasBeenInvoked() { return mAddNocCommandHasBeenInvoked; }
+    inline bool UpdateNocCommandHasBeenInvoked() { return mUpdateNocCommandHasBeenInvoked; }
 
-    inline void SetNocCommandInvoked(FabricIndex nocFabricIndex)
+    inline void SetAddNocCommandInvoked(FabricIndex nocFabricIndex)
     {
-        mNocCommandHasBeenInvoked = true;
-        mFabricIndex              = nocFabricIndex;
+        mAddNocCommandHasBeenInvoked = true;
```

Comment:
OK, so if the failsafe timer expires this will stay true?  And then the next failsafe context that gets set up will start off with it being true?

Reply:
Acked, we need to have mAddNocCommandHasBeenInvoked stay true for cleanup, ArmFailSafe command will always reset mAddNocCommandHasBeenInvoked and mNocCommandHasBeenInvoked